processor/spanmetricsprocessor/processor.go

```diff
@@ -410,11 +445,13 @@ func (p *processorImp) aggregateMetricsForSpan(serviceName string, span pdata.Sp
 	p.updateLatencyMetrics(key, latencyInMilliseconds, index)
 	p.updateLatencyExemplars(key, latencyInMilliseconds, span.TraceID())
 	p.lock.Unlock()
+	stats.Record(context.TODO(), spanProcessedCount.M(int64(1))) // count 1 here for a processed span.
```

Comment:
`context.TODO()` should never be used. This metric is incorrect, it's tracking how many `pdata.Traces` have been processed, not how many spans. I don't think this is useful as is.

Reply:
I thought it was counting spans because the function ```func (p *processorImp) aggregateMetricsForSpan(serviceName string, span pdata.Span, resourceAttr pdata.AttributeMap) {```  is being called inside the loop iterating on spans:
```
for k := 0; k < spans.Len(); k++ {
			span := spans.At(k)
			if filterspan.SkipSpan(p.include, p.exclude, span, rspans.Resource(), ils.InstrumentationLibrary()) {
				continue
			}
			p.aggregateMetricsForSpan(serviceName, span, rspans.Resource().Attributes())
		}
```

But I am going to move the counter outside the loop https://github.com/Shopify/opentelemetry-collector-contrib/blob/14b7a428a85735ad20536cbf686bbf225fca3ba5/processor/spanmetricsprocessor/processor.go#L424